core/src/common.cpp

```diff
@@ -362,6 +364,264 @@ namespace irods::http
+		auto tcp_stream{irods::http::transport_factory(url.scheme_id(), io_ctx)};
+		tcp_stream->connect(url.host(), *port);
+
+		// Build Request
+		constexpr auto http_version_number{11};
+		beast::http::request<beast::http::string_body> req{beast::http::verb::get, url.path(), http_version_number};
+		req.set(beast::http::field::host, irods::http::create_host_field(url, *port));
+		req.set(beast::http::field::user_agent, irods::http::version::server_name);
+		req.set(beast::http::field::accept, "application/json");
+		req.prepare_payload();
+
+		// Send request & receive response
+		auto res{tcp_stream->communicate(req)};
+
+		logging::debug("{}: Received the following response: [{}]", __func__, res.body());
+
+		// JSONize response
+		return res.body();
+	}
+
+	/// Adds the specified algorithm \p _alg to the verifier \p _verifier, using the additional informaiton provided by the JWK \p _jwk.
+	///
+	/// See RFC 7518 for details on JSON Web Algorithms (JWA)
+	///
+	/// \param[in,out] _verifier The jwt::verifier to add additional verification algorithms to.
+	/// \param[in]     _jwk      The JWK containing the required JWA information.
+	/// \param[in]     _alg      The sigining algorithm requested by the signed JWT.
+	///
+	/// \todo Should this be separated out by algorithm familiy?
+	auto add_alg_from_jwk(jwt::verifier<jwt::default_clock, jwt::traits::nlohmann_json>& _verifier, const jwt::jwk<jwt::traits::nlohmann_json>& _jwk, std::string_view _alg) -> void {
+		namespace logging = irods::http::log;
+
+		if (_alg == "RS256") {
+			logging::trace("{}: Detected [RS256], attempting extraction of attributes from JWK...", __func__);
+
+			// Get modulus parameter (JWA)
+			auto mod{_jwk.get_jwk_claim("n").as_string()};
+
+			// Get exponent parameter (JWA)
+			auto exp{_jwk.get_jwk_claim("e").as_string()};
+
+			// Add verification algorithm
+			_verifier.allow_algorithm(jwt::algorithm::rs256(jwt::helper::create_public_key_from_rsa_components(mod, exp)));
+		}
+		// For symetric excryption, no parameter 'k' is provided
+		// See OpenID Connect Discovery section 3
+		if (_alg == "HS256") {
+			logging::trace("{}: Detected [HS256], attempting extraction of attributes from JWK...", __func__);
+
+			// Attempt to get key value parameter (JWA)
+			auto key{_jwk.get_jwk_claim("k").as_string()};
+
+			// Add verification algorithm
+			_verifier.allow_algorithm(jwt::algorithm::hs256(key));
+		}
+		if (_alg == "ES256") {
+			logging::trace("{}: Detected [ES256], attempting extraction of attributes from JWK...", __func__);
+
+			// Get curve parameter (JWA)
+			auto crv{_jwk.get_jwk_claim("crv").as_string()};
+
+			// Get x coordinate parameter (JWA)
+			auto x{_jwk.get_jwk_claim("x").as_string()};
+
+			// Get y coordinate parameter (JWA)
+			auto y{_jwk.get_jwk_claim("y").as_string()};
+
+			// Add verification algorithm
+			_verifier.allow_algorithm(jwt::algorithm::es256(jwt::helper::create_public_key_from_ec_components(crv, x, y)));
+		}
```

Comment:
Given the params for a family of algorithms are the same, get the family, extract the params, then figure out the algorithm?

Reply:
Done.